src/ddmd/func.d

```diff
@@ -3864,71 +3864,48 @@ extern (C++) class FuncDeclaration : Declaration
  * Input:
  *      ad      aggregate with the invariant
  *      vthis   variable with 'this'
- *      direct  call invariant directly
  * Returns:
  *      void expression that calls the invariant
  */
-extern (C++) Expression addInvariant(Loc loc, Scope* sc, AggregateDeclaration ad, VarDeclaration vthis, bool direct)
+extern (C++) Expression addInvariant(Loc loc, Scope* sc, AggregateDeclaration ad, VarDeclaration vthis)
 {
     Expression e = null;
-    if (direct)
-    {
-        // Call invariant directly only if it exists
-        FuncDeclaration inv = ad.inv;
-        ClassDeclaration cd = ad.isClassDeclaration();
-
-        while (!inv && cd)
-        {
-            cd = cd.baseClass;
-            if (!cd)
-                break;
-            inv = cd.inv;
-        }
-        if (inv)
-        {
-            version (all)
-            {
-                // Workaround for https://issues.dlang.org/show_bug.cgi?id=13394
-                // For the correct mangling,
-                // run attribute inference on inv if needed.
-                inv.functionSemantic();
-            }
-
-            //e = new DsymbolExp(Loc(), inv);
-            //e = new CallExp(Loc(), e);
-            //e = e.semantic(sc2);
+    // Call invariant directly only if it exists
+    FuncDeclaration inv = ad.inv;
+    ClassDeclaration cd = ad.isClassDeclaration();
 
-            /* https://issues.dlang.org/show_bug.cgi?id=13113
-             * Currently virtual invariant calls completely
-             * bypass attribute enforcement.
-             * Change the behavior of pre-invariant call by following it.
-             */
-            e = new ThisExp(Loc());
-            e.type = vthis.type;
-            e = new DotVarExp(Loc(), e, inv, false);
-            e.type = inv.type;
-            e = new CallExp(Loc(), e);
-            e.type = Type.tvoid;
-        }
+    while (!inv && cd)
+    {
+        cd = cd.baseClass;
+        if (!cd)
+            break;
+        inv = cd.inv;
     }
-    else
+    if (inv)
     {
         version (all)
         {
             // Workaround for https://issues.dlang.org/show_bug.cgi?id=13394
             // For the correct mangling,
             // run attribute inference on inv if needed.
-            if (ad.isStructDeclaration() && ad.inv)
-                ad.inv.functionSemantic();
-        }
-        // Call invariant virtually
-        Expression v = new ThisExp(Loc());
-        v.type = vthis.type;
-        if (ad.isStructDeclaration())
-            v = v.addressOf();
-        e = new StringExp(Loc(), cast(char*)"null this");
-        e = new AssertExp(loc, v, e);
-        e = e.semantic(sc);
+            inv.functionSemantic();
+        }
+
+        //e = new DsymbolExp(Loc(), inv);
+        //e = new CallExp(Loc(), e);
+        //e = e.semantic(sc2);
```

Comment:
That was there before (it only shows up in the diff because it is de-indented). I can do that, but I'm not sure why it was there in the first place or what purpose it originally served (illustrative?).